l1-contracts/test/Inbox.t.sol

```diff
@@ -0,0 +1,241 @@
+import {IMessageBox} from "@aztec/core/interfaces/messagebridge/IMessageBox.sol";
+import {MessageBox} from "@aztec/core/messagebridge/MessageBox.sol";
+import {Registry} from "@aztec/core/messagebridge/Registry.sol";
+
+import {DataStructures} from "@aztec/core/libraries/DataStructures.sol";
+
+contract InboxTest is Test {
+  Inbox inbox;
+
+  event MessageAdded(
+    bytes32 indexed entryKey,
+    address indexed sender,
+    bytes32 indexed recipient,
+    uint256 senderChainId,
+    uint256 recipientVersion,
+    uint32 deadline,
+    uint64 fee,
+    bytes32 content
+  );
+  event L1ToL2MessageCancelled(bytes32 indexed entryKey);
+
+  function setUp() public {
+    address rollup = address(this);
+    Registry registry = new Registry();
+    inbox = new Inbox(address(registry));
+    registry.setAddresses(rollup, address(inbox), address(0x0));
+  }
+
+  function _helper_computeEntryKey(DataStructures.L1ToL2Msg memory message)
+    internal
+    pure
+    returns (bytes32)
+  {
+    return bytes32(
+      uint256(
+        sha256(
+          abi.encode(
+            message.sender,
+            message.recipient,
+            message.content,
+            message.secretHash,
+            message.deadline,
+            message.fee
+          )
+        )
+      ) % 21888242871839275222246405745257275088548364400416034343698204186575808495617
+    );
+  }
+
+  function _fakeMessage() internal view returns (DataStructures.L1ToL2Msg memory) {
+    return DataStructures.L1ToL2Msg({
+      sender: DataStructures.L1Actor({actor: address(this), chainId: block.chainid}),
+      recipient: DataStructures.L2Actor({
+        actor: 0x2000000000000000000000000000000000000000000000000000000000000000,
+        version: 1
+      }),
+      content: 0x3000000000000000000000000000000000000000000000000000000000000000,
+      secretHash: 0x4000000000000000000000000000000000000000000000000000000000000000,
+      fee: 5,
+      deadline: uint32(block.timestamp + 100)
+    });
+  }
+
+  function testFuzzSendL2Msg(DataStructures.L1ToL2Msg memory message) public {
+    // fix message.sender and deadline:
+    message.sender = DataStructures.L1Actor({actor: address(this), chainId: block.chainid});
+    if (message.deadline <= block.timestamp) {
+      message.deadline = uint32(block.timestamp + 100);
+    }
+    bytes32 expectedEntryKey = _helper_computeEntryKey(message);
+    vm.expectEmit(true, true, true, true);
+    // event we expect
+    emit MessageAdded(
+      expectedEntryKey,
+      message.sender.actor,
+      message.recipient.actor,
+      message.sender.chainId,
+      message.recipient.version,
+      message.deadline,
+      message.fee,
+      message.content
+    );
+    // event we will get
+    bytes32 entryKey = inbox.sendL2Message{value: message.fee}(
+      message.recipient, message.deadline, message.content, message.secretHash
+    );
+    assertEq(entryKey, expectedEntryKey);
+    assertEq(inbox.get(entryKey).count, 1);
```

Comment:
You don't need to fetch the entry 3 times here, you can just cache it in a var and check the 3 members.